### DISCUSSION TITLE
Robot's with no vision system knows position and direction in very high accuracy, what's sensor type installed in floor, and does Rtabmap support this type of sensors ?

### DISCUSSION THREAD

**Discussion by damavand1** [introlab/rtabmap#1197] [General]
[https://www.youtube.com/watch?v=IZbTk6YCQwQ](https://www.youtube.com/watch?v=IZbTk6YCQwQ)
[https://www.youtube.com/watch?v=j60X60SgPwg](https://www.youtube.com/watch?v=j60X60SgPwg)

![Screenshot 2024-01-07 172901](https://github.com/introlab/rtabmap/assets/104978921/088b5522-28ac-4836-88f6-f0620eb5445d)

as you see this robots knows the location and direction in very high accuracy
and they have no vision system

I know some kind of sensor installed on the floor, and a reader installed on robot 
What kind of sensor do you think this robot's use ?

and does Rtabmap support this sensors for localization ?

tnx

**Comment by matlabbe** [comment]
You don't need a SLAM system if you can get accurate global localization.

For that particular robotics solution, it could be a combination of odometry and unique marker detection through the floor indicating which cell the robot is in the floor matrix and recorrect each time the small odometry drift. Other option could be a magnetic rail under the floor.

**Reply by damavand1** [reply]
what unique marker detection method do you referee ?
what's the sensor name do you referee ?

sorry this question is far about Rtabmap

**Reply by matlabbe** [reply]
I don't have a particular sensor product in mind, but there are some sensors working like Apriltag detection by looking at a specified "qr code" on the floor to get a unique ID and relative position from the sensor on the robot (so that we can correct the odometry drift). In older videos, they used something similar to that: 
https://www.youtube.com/watch?v=hbyp6JE2f34
https://www.youtube.com/watch?v=JtdGpLsT4bE
https://www.youtube.com/watch?v=-99bGT0JtAU

In that application, the floor is black, either something else can give it from under it with some technology or the "tags" are not visible by eye (maybe an IR tag). It could also be just ceiling cameras looking down and tracking all the robots, though from the video, it is difficult to see the orientation of the robot from the top.

Just found this patent: https://patents.google.com/patent/EP3637213A1/en?inventor=Libiao&oq=Libiao their "magic carpet" uses magnetic poles that can be perceived by the robot with Hall sensors. If all squares are identical, I guess the robot has to enter the arena at a known place from the system, then just count the number of square traversed to know where it is.

> The position and direction of the robot in a modular zone can be determined, as long as the signals of two magnetic strips having different polarities can be collected by the plurality of Hall sensors. The plurality of Hall sensors may be arranged in various ways, e.g., being arranged in a straight line, in an angular line, in a curve, in a circle, etc.

> Preferably, the plurality of Hall sensors are arranged in a form of square, and the address placemark identification device is located at the center of the square. When the Hall sensors are arranged in a form of square, they can cover a relatively large area, and can easily collect the signals of the magnetic strips. As the robot moves, variations may occur in the output signals of the plurality of Hall sensors, and in this way, it is possible to obtain the accurate position of the robot.
